dive_core/data_core.cpp

```diff
@@ -372,16 +393,303 @@ void CaptureMetadataCreator::FillViewportState(EventStateInfo::Iterator event_st
+            back.writeMask = rb_stencilwrmask.bitfields.BFWRMASK;
+        }
+        event_state_it->SetStencilOpStateFront(front);
+        event_state_it->SetStencilOpStateBack(back);
+    }
+}
 
 //--------------------------------------------------------------------------------------------------
-void CaptureMetadataCreator::FillColorBlendState(EventStateInfo::Iterator event_state_it) {}
+void CaptureMetadataCreator::FillColorBlendState(EventStateInfo::Iterator event_state_it)
+{
+    uint32_t rt_id = 0;
+    uint32_t rb_mrt_reg_start = GetRegOffsetByName("RB_MRT0_CONTROL");
+    DIVE_ASSERT(rb_mrt_reg_start != kInvalidRegOffset);
+    constexpr uint32_t kElemCount = 8;
+    while (rt_id < 8 && m_state_tracker.IsRegSet(rb_mrt_reg_start + kElemCount * rt_id))
+    {
+        RB_MRT_CONTROL       rb_mrt_control;
+        RB_MRT_BLEND_CONTROL rb_mrt_blend_control;
+
+        // Assumption: These are all set together. All-or-nothing.
+        const uint32_t mrt_reg = rb_mrt_reg_start + kElemCount * rt_id;
+        rb_mrt_control.u32All = m_state_tracker.GetRegValue(mrt_reg);
+        rb_mrt_blend_control.u32All = m_state_tracker.GetRegValue(mrt_reg + 1);
+
+        VkPipelineColorBlendAttachmentState attach;
+        attach.blendEnable = (rb_mrt_control.bitfields.BLEND == 1) ? VK_TRUE : VK_FALSE;
+
+        // Be careful!!! Here we assume the enum `VkBlendFactor` matches exactly
+        // `adreno_rb_blend_factor`
+        attach.srcColorBlendFactor = static_cast<VkBlendFactor>(
+        rb_mrt_blend_control.bitfields.RGB_SRC_FACTOR);
+        // Be careful!!! Here we assume the enum `VkBlendFactor` matches exactly
+        // `adreno_rb_blend_factor`
+        attach.dstColorBlendFactor = static_cast<VkBlendFactor>(
+        rb_mrt_blend_control.bitfields.RGB_DEST_FACTOR);
+        // Be careful!!! Here we assume the enum `VkBlendOp` matches exactly `a3xx_rb_blend_opcode`
+        attach.colorBlendOp = static_cast<VkBlendOp>(
+        rb_mrt_blend_control.bitfields.RGB_BLEND_OPCODE);
+        // Be careful!!! Here we assume the enum `VkBlendFactor` matches exactly
+        // `adreno_rb_blend_factor`
+        attach.srcAlphaBlendFactor = static_cast<VkBlendFactor>(
+        rb_mrt_blend_control.bitfields.ALPHA_SRC_FACTOR);
+        // Be careful!!! Here we assume the enum `VkBlendFactor` matches exactly
+        // `adreno_rb_blend_factor`
+        attach.dstAlphaBlendFactor = static_cast<VkBlendFactor>(
+        rb_mrt_blend_control.bitfields.ALPHA_DEST_FACTOR);
+        // Be careful!!! Here we assume the enum `VkBlendOp` matches exactly `a3xx_rb_blend_opcode`
+        attach.alphaBlendOp = static_cast<VkBlendOp>(
+        rb_mrt_blend_control.bitfields.ALPHA_BLEND_OPCODE);
+        event_state_it->SetAttachment(rt_id, attach);
+
+        ++rt_id;
+    }
+
+    // TODO(wangra): rop code should be set per MR
+    // if (m_state_tracker.IsContextStateSet(mmCB_COLOR_CONTROL))
+    //{
```

Comment:
yes, i left them here on purpose. I think the rop code should be set per MR whereas the original Dive code only set the 1st one. I will revisit this once we have the DRAW_STATES implemented so that i could verify this.